addons/sale/models/sale_order.py

```diff
@@ -329,7 +333,7 @@ def _compute_expected_date(self):
                 dt = line._expected_date()
                 dates_list.append(dt)
             if dates_list:
-                order.expected_date = fields.Datetime.to_string(min(dates_list))
+                order.expected_date = Datetime.to_string(min(dates_list))
```

Comment:
```suggestion
                order.expected_date = min(dates_list)
```
str to datetime conversion isn't necessary anymore IMHO

Reply:
:+1: especially since it will be converted back to a `date` when put in cache.